### PR TITLE
TIP-1262 Turn Tool commands into services (exit ContainerAwareCommand)

### PR DESCRIPTION
**Pull request by arnolanglade**
**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Tool/Bundle/ElasticsearchBundle/Resources/config/services.yml [hunk 1]
```diff
@@ -8,3 +8,9 @@ services:
 
     akeneo_elasticsearch.client_builder:
         class: '%akeneo_elasticsearch.client_builder.class%'
+
+    Akeneo\Tool\Bundle\ElasticsearchBundle\Command\ResetIndexesCommand:
+        arguments:
+            - '@akeneo_elasticsearch.registry.clients'
+        tags:
+            - { name: console.command }
```
Comment (BitOne) [inline src/Akeneo/Tool/Bundle/ElasticsearchBundle/Resources/config/services.yml]: Missing new line at end of file.
IDE  not properly configured? :tractor:

File: src/Akeneo/Tool/Bundle/FileStorageBundle/Resources/config/commands.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+services:
+    Akeneo\Tool\Bundle\FileStorageBundle\Command\StoreFileCommand:
+        arguments:
+            - '@akeneo_file_storage.file_storage.file.file_storer'
+            - '@oneup_flysystem.mount_manager'
+        tags:
+            - { name: console.command }
```
Comment (BitOne) [inline src/Akeneo/Tool/Bundle/FileStorageBundle/Resources/config/commands.yml]: Again :tractor:

File: src/Akeneo/Tool/Bundle/VersioningBundle/Resources/config/commands.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+            - { name: console.command }
+
+    Akeneo\Tool\Bundle\VersioningBundle\Command\RefreshCommand:
+        arguments:
+            - '@logger'
+            - '@pim_versioning.manager.version'
+            - '@akeneo_storage_utils.doctrine.object_detacher'
+            - '@doctrine.orm.default_entity_manager'
+        tags:
+            - { name: console.command }
```
Comment (BitOne) [inline src/Akeneo/Tool/Bundle/VersioningBundle/Resources/config/commands.yml]: Here as well ;) :tractor:

File: src/Akeneo/Tool/Bundle/VersioningBundle/Resources/config/commands.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+services:
```
Comment (ahocquard) [inline src/Akeneo/Tool/Bundle/VersioningBundle/Resources/config/commands.yml]: cli_commands.yml would be better as filename (command is used for several things).

Comment (arnolanglade) [reply src/Akeneo/Tool/Bundle/VersioningBundle/Resources/config/commands.yml]: if you want :) but there are lot of commands declared as a services in file named `commands.yml`. I don't think this is a good idea by the way! Those service declarations should go files named with business words (like `product.yml`).

File: src/Akeneo/Tool/Bundle/FileStorageBundle/Command/StoreFileCommand.php [hunk 5]
```diff
@@ -50,8 +68,7 @@ protected function execute(InputInterface $input, OutputInterface $output)
         }
 
         $rawFile = new \SplFileInfo($filePath);
-        $storer = $this->getFileStorer();
```
Comment (nmarniesse) [inline src/Akeneo/Tool/Bundle/FileStorageBundle/Command/StoreFileCommand.php]: `getFileStorer()` method is not removed

File: src/Akeneo/Tool/Bundle/BatchQueueBundle/Command/JobQueueConsumerCommand.php [hunk 6]
```diff
@@ -28,26 +28,53 @@
  * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
  */
-class JobQueueConsumerCommand extends ContainerAwareCommand
+class JobQueueConsumerCommand extends Command
 {
+    public const COMMAND_NAME = 'akeneo:batch:job-queue-consumer-daemon';
```
Comment (pierallard) [inline src/Akeneo/Tool/Bundle/BatchQueueBundle/Command/JobQueueConsumerCommand.php]: if not used, remove (I can't see usage)

Comment (arnolanglade) [reply src/Akeneo/Tool/Bundle/BatchQueueBundle/Command/JobQueueConsumerCommand.php]: it is in other classes.

**Pull request merged**